modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/partition.py

```diff
@@ -302,7 +301,7 @@ def ip(self, materialize=True):
             if len(self.call_queue):
                 self.drain_call_queue()
             else:
-                self._ip_cache = self.apply(lambda df: pandas.DataFrame([]))._ip_cache
+                self._ip_cache = self.apply(lambda df: None)._ip_cache
```

Comment:
On a cluster, this may not be correct, since if all the processes of one node are busy, then the engine can start the calculation on another node, which will lead to incorrect receipt of the address.

One of the most likely solutions would be to return the entire dataframe and completely update the partition (all references to remote objects).

It may potentially be possible to use special API engines to obtain the address more efficiently. (for example, like in https://github.com/ray-project/ray/issues/13630)

@YarShev thanks for the explanation and solution proposal!